dtfs-central-api/src/v1/validation/utilisation-report-service/selected-fee-record-validator.ts

```diff
@@ -10,3 +11,24 @@ export const validateSelectedFeeRecordsAllHaveSamePaymentCurrency = (selectedFee
     throw new InvalidPayloadError('Selected fee records must all have the same payment currency');
   }
 };
+
+/**
+ * Validates that all selected fee records with payments have the same fee
+ * record payment group and that the fee record IDs match the fee record IDs on
+ * this existing payment group.
+ * @param feeRecordIds - The fee record IDs
+ */
+export const validateThatAllSelectedFeeRecordsWithPaymentsFormACompletePaymentGroup = async (feeRecordIds: number[]) => {
+  const selectedFeeRecords = await FeeRecordRepo.findByIdWithPaymentsAndFeeRecords(feeRecordIds);
+
+  const firstFeeRecord = selectedFeeRecords[0];
+
+  if (firstFeeRecord.status === FEE_RECORD_STATUS.DOES_NOT_MATCH) {
+    const firstPayment = firstFeeRecord.payments[0];
+    const savedFeeRecordIds = firstPayment.feeRecords.map((feeRecord) => feeRecord.id);
+
+    if (!savedFeeRecordIds.every((id) => feeRecordIds.includes(id)) || !feeRecordIds.every((id) => savedFeeRecordIds.includes(id))) {
+      throw new InvalidPayloadError('Requested fee record IDs do not match the fee record IDs on the existing payment group.');
+    }
+  }
```

Comment:
Would it look better if we 

```suggestion
  if (firstFeeRecord.status !== FEE_RECORD_STATUS.DOES_NOT_MATCH) {
    return;
  }
    const firstPayment = firstFeeRecord.payments[0];
    const savedFeeRecordIds = firstPayment.feeRecords.map((feeRecord) => feeRecord.id);

    const allSavedFeeRecordIdsArePresentInProvidedFeeRecords = savedFeeRecordIds.every((id) => feeRecordIds.includes(id));
    const allProvidedIdsArePresentInSavedFeeRecords = feeRecordIds.every((id) => savedFeeRecordIds.includes(id));

    if (!allSavedFeeRecordIdsArePresentInProvidedFeeRecords || !allProvidedIdsArePresentInSavedFeeRecords) {
      throw new InvalidPayloadError('Requested fee record IDs do not match the fee record IDs on the existing payment group.');
  }
```

Reply:
You could actually optimise this even further to avoid looping twice, but might be at the cost of readability so this is just a general comment and no action required


```suggestion
  if (firstFeeRecord.status !== FEE_RECORD_STATUS.DOES_NOT_MATCH) {  
    return;  
  }  
    const firstPayment = firstFeeRecord.payments[0];  
    const savedFeeRecordIds = firstPayment.feeRecords.map((feeRecord) => feeRecord.id); // sets would be faster for large volumes but not seen/needed here

    const allIdsMatch = feeRecordIds.every((id) => savedFeeRecordIds.includes(id)) && savedFeeRecordIds.size === feeRecordIds.length;

    if (  
        !allIdsMatch
    ) {  
      throw new InvalidPayloadError('Requested fee record IDs do not match the fee record IDs on the existing payment group.');  
  }  
```

Reply:
Thanks, updated this